gossipd/gossip_wire.csv

```diff
@@ -13,15 +13,16 @@ gossipctl_init,,lflen,u16
 gossipctl_init,,lfeatures,lflen*u8
 gossipctl_init,,num_wireaddrs,u16
 gossipctl_init,,wireaddrs,num_wireaddrs*struct wireaddr
+gossipctl_init,,announce_listen,num_wireaddrs*u32
```

Comment:
While `u32` on most systems using gcc will match `enum`, C99 standard 6.7.2.2 is:

> Each enumerated type shall be compatible with `char`, a signed integer type, or an unsigned integer type. The choice of type is implementation-defined, but shall be capable of representing the values of all the members of the enumeration.

This may bite us later.  Under gcc `sizeof(enum foo) == sizeof(int)` and in most systems `sizeof(int) == sizeof(uint32_t)`.

Reply:
Yeah, that was kind of a hack (though you'd get a compiler warning if it was wrong).  We can actually provide proper mashalling for this type.

Reply:
Also nit: you consistently use `listen_announce` elsewhere, and only this csv file uses `announce_listen`.